konig-workbook-update-maven-plugin/src/main/java/io/konig/maven/google/update/GoogleSheetUpdater.java

```diff
@@ -0,0 +1,1119 @@
+	private static final String DECIMAL_SCALE = "Decimal Scale";
+	private static final int HEADER_ROW = 1;
+	private Map<URI,String> prefixList = new HashMap<>();
+	private Map<String,String> targetClassList = new HashMap<>();
+	private ValueFactory vf = new ValueFactoryImpl();
+	
+	private boolean isOverrideSheetValues = false;
+	private boolean isOverrideRDFValues = false;
+	private boolean isRaiseError = false;
+	
+	private  Sheets sheet = null;
+	private  String sheetId = null;
+	
+	public GoogleSheetUpdater(String option) {
+		sheet = getSheetService();
+		if(option.equals("OPTION 1")) {
+			isOverrideSheetValues = true;
+		} else if (option.equals("OPTION 2")) {
+			isOverrideRDFValues = true;
+		} else {
+			isRaiseError = true;
+		} 
+	}
+	
+	public void execute(FileSet[] owlFileSet ,FileSet[] shaclFileSet, String documentId) throws RDFParseException, RDFHandlerException, IOException, IriTemplateParseException {
+		this.sheetId = documentId;
+		for(FileSet fileSet : owlFileSet) {
+			if(fileSet.getDirectory() != null){
+				File viewDir = new File(fileSet.getDirectory());
+				File[] files = viewDir.listFiles();
+				for(File file : files) {
+					try (InputStream inputStream = new FileInputStream(file)) {
+						Graph graph = new MemoryGraph();
+						RdfUtil.loadTurtle(graph, inputStream, "");
+						updateOntology(graph);
+						//updateClasses(graph);
+						//updateProperties(graph);
+					}
+				}
+			}
+		}
+		for(FileSet fileSet : shaclFileSet) {
+			if(fileSet.getDirectory() != null){
+				File viewDir = new File(fileSet.getDirectory());
+				File[] files = viewDir.listFiles();
+				for(File file : files) {
+					try (InputStream inputStream = new FileInputStream(file)) {
+						Graph graph = new MemoryGraph();
+						RdfUtil.loadTurtle(graph, inputStream, "");
+						updateShapes(graph);
+						updatePropertyConstraints(graph);
+					}
+				}
+			}
+		}
+	}
+	
+	private Sheets getSheetService() {
+		String credentialFile = System.getProperty("io.konig.maven.google.update");
+		if(credentialFile == null || credentialFile.equals("")) {
+			throw new KonigException("Credential not found. Please define the 'io.konig.maven.google.update' system property ");
+		}
+		InputStream in = null;
+		try {
+			in = new FileInputStream(new File(credentialFile));
+			GoogleCredential credential = GoogleCredential.fromStream(in).createScoped(DriveScopes.all());
+			sheet = new Sheets.Builder(Utils.getDefaultTransport(), Utils.getDefaultJsonFactory(), credential)
+					.setApplicationName("Google Sheet API")
+					.build();
+		}catch(Exception ex){
+			throw new KonigException(ex);
+		}
+		return sheet;
+	}
+	
+	private void updateOntology(Graph graph) throws IOException {
+		List<Vertex> ontologyList = graph.v(OWL.ONTOLOGY).in(RDF.TYPE).toVertexList();
+		List<List<Object>> ontologies = null;
+		List<Object> headers = new ArrayList<>();
+		int columns = 0;
+		
+		if(isSheetExist(SHEET_ONTOLOGY)) {
```

Comment:
Sheet names are not important in a semantic workbook.
There is no requirement that ontology descriptions be placed in a sheet named "Ontologies", no requirement that class descriptions be placed in a sheet named "Classes", and so on.

Indeed, it is possible to have information about a given type of entity (such as `Ontology`) spread across multiple sheets with arbitrary names.  The solution must identify the type of sheet based on a unique set of required columns.

For instance, the column named `Ontology Name` is unique to a sheet that contains Ontology descriptions.   Your solution should scan all sheets and classify them based on the unique set of required columns.

This design introduces a couple challenges: 

1.  You must search through all sheets of a given type to see if there is a match for the individual that needs to be updated.
2.  If there are multiple sheets for a given type of entity, how do we decide which sheet to use when adding new entities of a given type?  Let's adopt a policy of adding new individuals to the first sheet containing entities of the given type.